frontend/javascripts/oxalis/view/right-border-tabs/comment_tab/comment_tab_view.tsx

```diff
@@ -363,200 +322,197 @@ class CommentTabView extends React.Component<PropsWithSkeleton, CommentTabState>
+                </AdvancedSearchPopover>
+                <ButtonComponent
+                  title="Jump to previous comment"
+                  onClick={previousComment}
+                  icon={<ArrowLeftOutlined />}
+                />
+                <InputComponent
+                  value={activeCommentContent}
+                  disabled={isEditingDisabled}
+                  title={allowUpdate ? undefined : messages["tracing.read_only_mode_notification"]}
+                  onChange={(evt: React.ChangeEvent<HTMLInputElement>) =>
+                    handleChangeInput(evt.target.value, true)
+                  }
+                  onPressEnter={(evt: React.KeyboardEvent<HTMLInputElement>) =>
+                    (evt.target as HTMLElement).blur()
+                  }
+                  placeholder="Add comment"
+                  style={{
+                    width: "50%",
+                  }}
+                />
+                <ButtonComponent
+                  onClick={() => setMarkdownModalVisibility(true)}
+                  disabled={isEditingDisabled}
+                  title={
+                    allowUpdate
+                      ? "Open dialog to edit comment in multi-line mode"
+                      : messages["tracing.read_only_mode_notification"]
+                  }
+                  type={isMultilineComment ? "primary" : "default"}
+                  icon={<EditOutlined />}
+                />
+                <ButtonComponent
+                  title="Jump to next comment"
+                  onClick={() => nextComment()}
+                  icon={<ArrowRightOutlined />}
+                />
+                <Dropdown menu={getSortDropdown()} trigger={["click"]}>
+                  <ButtonComponent title="Sort" onClick={toggleSortingDirection}>
+                    {renderSortIcon()}
+                  </ButtonComponent>
+                </Dropdown>
+                <ButtonComponent
+                  onClick={toggleExpandAllTrees}
+                  icon={<ShrinkOutlined />}
+                  title="Collapse or expand groups"
+                />
+              </Space.Compact>
+              <div
+                style={{
+                  flex: "1 1 auto",
+                  marginTop: 20,
+                  listStyle: "none",
+                }}
+              >
+                {renderCommentTree()}
+              </div>
+            </React.Fragment>
+          );
+        }}
+      </DomVisibilityObserver>
+    </div>
+  );
 }
 
-const mapStateToProps = (state: OxalisState) => ({
-  skeletonTracing: state.tracing.skeleton,
-  allowUpdate: state.tracing.restrictions.allowUpdate,
-});
+const CommentTabViewMemo = React.memo(
+  CommentTabView,
+  function arePropsEqual(prevPops: Props, nextProps: Props) {
+    if (prevPops.skeletonTracing.activeNodeId !== nextProps.skeletonTracing.activeNodeId) {
+      return false;
+    }
```

Comment:
good catch

Reply:
Nope, sorry forgot to delete this comment:

I just checked and wk increases the node id with each node, regardless of what the tree is. Therefore, having duplicate node ids shouldn't be possible and thus checking for the changed active tree shouldn't be necessary.

Or can you think of a way to have duplicate node ids in an annotation (e.g. importing an specially crafted nml) @philippotto?

Reply:
node ids should always be unique and we take care of this. if there aren't we will have several other problems :sweat_smile: 

> e.g. importing an specially crafted nml

the frontend guards against this.

returning back to the actual question:
comparing active node id **and** active tree id is a bit safer, because both are nullable. this means, theoretically (not sure whether currently easy to trigger), the active tree id could change while the active node id stays null.

Reply:
I restored the active tree check with commit 1a55f10a0067c94f1ae10b171d551462be2a7044

Reply:
> comparing active node id and active tree id is a bit safer,

Then, could you please re-add the check for the active tree change?